src/components/RouletteMain/RouletteMain.jsx

```diff
@@ -0,0 +1,151 @@
+import React, { Component } from 'react'
+import Header from '../Header'
+import Footer from '../Footer'
+import ErrorBoundary from '../ErrorBoundary'
+import { ProcessGetRequest } from '../../utils/RestUtils'
+import EditMovie from '../EditMovie'
+import AddMovie from '../AddMovie'
+import DeleteMovie from '../DeleteMovie'
+import SortListing from '../SortListing'
+import "regenerator-runtime"
+const MoviesList = React.lazy(() => {
+    return new Promise(resolve => setTimeout(resolve, 500)).then(() => import("../MoviesList"));
+})
+
+class RouletteMain extends Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            movieList: [],
+            sortOption: "year", 
+            isError: false,
+            isAddModalVisible: false,
+            isEditModalVisible: false,
+            isDeleteModalVisible: false,
+            processEditMovieField: {},
+            processDeleteMovieField: {}
+        }
+    }
+
+    onAddAction() {
+        this.setState({ isAddModalVisible: true })
+    }
+
+    onEditAction(id) {
+        let stateList = this.state.movieList;
+        stateList.forEach((obj, i) => { if (obj.id === id) { this.setState({ processEditMovieField: obj }) } });
+        this.setState({ isEditModalVisible: true })
+    }
+
+    onDeleteAction(id) {
+        let stateList = this.state.movieList;
+        stateList.forEach((obj, i) => { if (obj.id === id) { this.setState({ processDeleteMovieField: obj }) } });
+        this.setState({ isDeleteModalVisible: true })
+    }
+
+    onCloseAction() {
+        this.setState({ isAddModalVisible: false, isEditModalVisible: false, isDeleteModalVisible: false })
+    }
+
+    onSubmitAction(item) {
+        let stateList = this.state.movieList;
+        let uniqueId = stateList.length + 1;
+        item.id = uniqueId;
+        stateList.push(item);
```

Comment:
Avoid state mutations, instead of this, please, use spread syntax: ```this.setState({ movieList: [...this.state.movieList, item] })```. For example, if you extend component from the React.PureComponent, remove **this.onCloseAction();** string, list of movie will not be updated because **stateList** and **this.state.movieList** are same.

Reply:
Fixed. Pushing and merging changes to master.